pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -607,14 +607,22 @@ ASTFormalParameter FormalParameter()  :
    (
 	simpleNode = ID()
 	// the order of outer "|" is important !
-	( LOOKAHEAD(2) ( LOOKAHEAD(2) ( (<OUT>|(<IN> <OUT>)) (LOOKAHEAD(2) <NOCOPY>  )? ) | <IN> ) )?
+    (LOOKAHEAD (2) (<IN>) FormalParameterMode())?
+    (LOOKAHEAD (2) (<OUT>) FormalParameterMode())?
+    (LOOKAHEAD (3) (<NOCOPY>) FormalParameterMode())?
 	("..."    | Datatype())
 	( (":" "="|<_DEFAULT>) Expression() )?
    )
    { jjtThis.setImage(simpleNode.getImage()) ;  return jjtThis ; }
 }
 
-
+ASTFormalParameterMode FormalParameterMode():
```

Comment:
Wouldn't it be more useful to enhance [ASTFormalParameter](https://github.com/pmd/pmd/blob/master/pmd-plsql/src/main/java/net/sourceforge/pmd/lang/plsql/ast/ASTFormalParameter.java)?

You could define a enum there for the "Mode" and set it above in lines 610-612: `<IN> jjtThis.setMode(ASTFormalParameter.Mode.IN)` and so on....

We should also make sure to cover this in tests, I don't know, whether `x IN OUT NUMBER` is still correctly recognized...

The class ASTFormalParameters can then have several getters, e.g. `isIn()`, `isInOut()`, `isNoCopy()` ... which makes this easily consumable in a XPath rule.

I don't think, these tokens justify to create a complete new AST node...